File: example06/visualstudio/dllExample06/dllExample06/InformationFactory.cpp
#include "pch.h"
#include "InformationFactory.hpp"

InformationFactory::InformationFactory()
{

}

InformationFactory::~InformationFactory()
{

}

IInformation* InformationFactory::newInformation(const char* text)
{
	IInformation* info = new Information();
	info->setInformation(text);
	return info;
}

File: example06/eclipse/testExample06/src/example06C/InformationFactory.java
package example06C;

public class InformationFactory
{
	public InformationFactory()
	{
		
	}
	
	public Information newInformation(String info) throws NativeCreationException
	{
		long pointer = newInformationNative(info);
		if (pointer != 0) return new Information(pointer);
		else throw new NativeCreationException();
	}
	
	// this method may be private in Java, but it's still accessible in C++
	private native long newInformationNative(String info);
}
